### PR TITLE
core: add multi-argument CV_Assert

### PR DESCRIPTION
**Pull request by sovrasov**

### CODE REVIEW

**Comment by alalek** [comment]
My suggestion is to stop using of assertions for parameters validation with unclear messages for end-users.

I believe we should introduce something like `CV_CheckArg` macros, including:
- `CV_CheckArg(condition_check, helper_mesage)`
- `CV_CheckArgEQ(value1, value2, helper_mesage)` with expansion of passed values. Probably we need pretty-printers to expand Mat types or expand macro set:
- `CV_CheckMatTypeEQ(value1, value2, helper_mesage)`
- `CV_CheckMatDepthEQ(value1, value2, helper_mesage)`
- `CV_CheckSizeEQ(value1, value2, helper_mesage)`
- something for multiple values check: line `channels() == 1 || channels() == 3`
- etc

File: modules/core/include/opencv2/core/base.hpp [hunk 1]
```diff
@@ -424,7 +424,22 @@ The macros CV_Assert (and CV_DbgAssert(expr)) evaluate the specified expression.
...
+#define CV_VA_NUM_ARGS(...)      CV_VA_NUM_ARGS_HELPER(__VA_ARGS__, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1)
+
+#define CV_Assert_1( expr ) if(!!(expr)) ; else cv::error( cv::Error::StsAssert, #expr, CV_Func, __FILE__, __LINE__ )
+#define CV_Assert_2( expr1, expr2 ) CV_Assert_1(expr1); CV_Assert_1(expr2)
+#define CV_Assert_3( expr1, expr2, expr3 ) CV_Assert_2(expr1, expr2); CV_Assert_1(expr3)
+#define CV_Assert_4( expr1, expr2, expr3, expr4 ) CV_Assert_3(expr1, expr2, expr3); CV_Assert_1(expr4)
+#define CV_Assert_5( expr1, expr2, expr3, expr4, expr5 ) CV_Assert_4(expr1, expr2, expr3, expr4); CV_Assert_1(expr5)
+#define CV_Assert_6( expr1, expr2, expr3, expr4, expr5, expr6 ) CV_Assert_5(expr1, expr2, expr3, expr4, expr5); CV_Assert_1(expr6)
+#define CV_Assert_7( expr1, expr2, expr3, expr4, expr5, expr6, expr7 ) CV_Assert_6(expr1, expr2, expr3, expr4, expr5, expr6 ); CV_Assert_1(expr7)
+#define CV_Assert_8( expr1, expr2, expr3, expr4, expr5, expr6, expr7, expr8 ) CV_Assert_6(expr1, expr2, expr3, expr4, expr5, expr6, expr7 ); CV_Assert_1(expr8)
```
Comment (csukuangfj) [inline modules/core/include/opencv2/core/base.hpp]: It should be CV_Assert_7.

File: modules/core/include/opencv2/core/base.hpp [hunk 2]
```diff
@@ -424,7 +424,22 @@ The macros CV_Assert (and CV_DbgAssert(expr)) evaluate the specified expression.
...
+#define CV_Assert_9( expr1, expr2, expr3, expr4, expr5, expr6, expr7, expr8, expr9 ) CV_Assert_6(expr1, expr2, expr3, expr4, expr5, expr6, expr7, expr8 ); CV_Assert_1(expr9)
```
Comment (csukuangfj) [inline modules/core/include/opencv2/core/base.hpp]: It should be CV_Assert_8.

File: modules/core/include/opencv2/core/base.hpp [hunk 3]
```diff
@@ -424,7 +424,22 @@ The macros CV_Assert (and CV_DbgAssert(expr)) evaluate the specified expression.
...
+#define CV_Assert_10( expr1, expr2, expr3, expr4, expr5, expr6, expr7, expr8, expr9, expr10 ) CV_Assert_6(expr1, expr2, expr3, expr4, expr5, expr6, expr7, expr8, expr9 ); CV_Assert_1(expr10)
```
Comment (csukuangfj) [inline modules/core/include/opencv2/core/base.hpp]: It should be CV_Assert_9.

**Comment by vpisarev** [comment]
ok, let's merge this PR with incremental improvement and then work on introducing better parameter check mechanisms. :+1:

**Pull request merged**